### ISSUE TITLE
Promise returned by setItem is resolved only after persist

### ISSUE THREAD

**Issue by citizen4371** [simonlast/node-persist#67]
Documentation states that 

> in the case where you set options.interval or options.continuous=false, your (optional) callback or your returned promise from this function will still get resolved immediately, even if the value has not been persisted to disk yet

but the promise is actually resolved only after persist is executed, so my question is - is it a bug, or is the documentation wrong ?

**Comment by akhoury** [comment]
@citizen4371 the docs look wrong, thanks for catching that.

By default - the `options.continuous` is `true`, this options basically means that the storage will always persist to disk every time `setItem()` gets called, HOWEVER, if you set `continuous` to` false`, it will NEVER persist to disk on its own, and will just hold the data in memory till either one those things happen:

* you manually call `storage.persist()` or `storage.persistSync()`
* OR if you have `options.interval = 60000 /* say 1 minute */` - this means that every 1 minute, wether you `persist()` or not, it will automatically persist to disk every time this interval handler is ran.

Now back to your original question, by default, if you don't change the `continuous` option, then your promise will resolve __after the persist to disk has been done__, but if you set `continuous` to `false` - both the __callback and the promise of `setItem()` will not be resolved, and will wait till you manually call `persist()` or the interval handler kicks in - which is like a fine grain control that was introduced originally in the library, I don't think it is very useful, but i left it in there.

However, now that I think about it, in the case of `continuous=false` I think I wanted to have the callback and the promise resolve immediately because I don't think it's reasonable for your promise to wait till you manually call `persist()` or till the configurable interval handler kicks in, so we __should__ resolve the `setItem` promise immediately, assuming you are handling the persist to disk manually.

## TL;DR

The documentation is wrong but it is what I want it to be, so I am going to say this is a bug.

**Comment by citizen4371** [comment]
Ok, got it, thanks for the quick reply.
